flow/src/main/java/com/sequenceiq/flow/domain/FlowLog.java

```diff
@@ -116,6 +120,35 @@ public FlowLog(Long resourceId,
         this.variablesJackson = variablesJackson;
         this.flowType = flowType;
         this.currentState = currentState;
+        this.endTime = null;
+    }
+
+    public FlowLog(Long resourceId,
+            String flowId,
+            String flowChainId,
+            String flowTriggerUserCrn,
+            String nextEvent,
+            String payload,
+            String payloadJackson,
+            ClassValue payloadType,
+            String variables,
+            String variablesJackson,
+            ClassValue flowType,
+            String currentState,
+            Long endTime) {
+        this.resourceId = resourceId;
+        this.flowId = flowId;
+        this.flowChainId = flowChainId;
+        this.flowTriggerUserCrn = flowTriggerUserCrn;
+        this.nextEvent = nextEvent;
+        this.payload = payload;
+        this.payloadJackson = payloadJackson;
+        this.payloadType = payloadType;
+        this.variables = variables;
+        this.variablesJackson = variablesJackson;
+        this.flowType = flowType;
+        this.currentState = currentState;
```

Comment:
Why not reuse the previous ctor?

Reply:
the previous constructor was being used and i didnt want to change any existing code

Reply:
Call with this and set the new parameter only.